pkg/helper/log/pipeline_reader.go

```diff
@@ -103,9 +103,9 @@ func (r *Reader) readAvailablePipelineLogs(pr *v1alpha1.PipelineRun) (<-chan Log
 		defer close(logC)
 		defer close(errC)
 
+		// clone the object to keep task number and name separately
+		c := r.clone()
```

Comment:
The copy object is used just for loop, not in goroutine.
so it is enough just one copy object for this loop.